File: src/main/java/com/machiav3lli/backup/activities/BaseActivity.java
package com.machiav3lli.backup.activities;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.preference.PreferenceManager;

import androidx.appcompat.app.AppCompatActivity;

import com.machiav3lli.backup.Constants;
import com.machiav3lli.backup.utils.PrefUtils;

public abstract class BaseActivity extends AppCompatActivity {

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(this);
        String langCode = prefs.getString(Constants.PREFS_LANGUAGES,
                Constants.PREFS_LANGUAGES_DEFAULT);
        assert langCode != null;
        PrefUtils.initLanguage(this, langCode);
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
    }

    @Override
    protected void onResume() {
        super.onResume();
    }

    @Override
    public void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
    }
}

File: app/src/main/java/com/machiav3lli/backup/activities/BaseActivity.kt
/*
 * OAndBackupX: open-source apps backup and restore app.
 * Copyright (C) 2020  Antonios Hazim
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */
package com.machiav3lli.backup.activities

import android.content.Context
import android.os.Bundle
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import com.machiav3lli.backup.ContextWrapperX.Companion.wrap
import com.machiav3lli.backup.OABX
import com.machiav3lli.backup.utils.TraceUtils
import com.machiav3lli.backup.utils.setCustomTheme
import timber.log.Timber

abstract class BaseActivity : AppCompatActivity() {

    override fun attachBaseContext(newBase: Context) {
        super.attachBaseContext(wrap(newBase))
    }

    override fun onCreate(savedInstanceState: Bundle?) {

        OABX.addActivity(this)

        setCustomTheme()
        enableEdgeToEdge()

        super.onCreate(savedInstanceState)

        Timber.w(
            "======================================== create ${
                TraceUtils.classAndId(this)
            }"
        )
    }

    override fun onResume() {
        Timber.w(
            "---------------------------------------- resume ${
                TraceUtils.classAndId(this)
            }"
        )
        OABX.resumeActivity(this)
        super.onResume()
    }

    override fun onDestroy() {
        Timber.w(
            "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ destroy ${
                TraceUtils.classAndId(this)
            }"
        )
        OABX.removeActivity(this)
        super.onDestroy()
    }
}